File: c++/leetcode/1317. Convert Integer to the Sum of Two No-Zero Integers.cpp
class Solution
{
public:
  vector<int> getNoZeroIntegers(int n)
  {
    int A = n, B = 0;

    const auto hasZero = [](int num) {
      if (num == 0)
        return true;

      while (num)
      {
        if ((num % 10) == 0)
          return true;
        num /= 10;
      }

      return false;
    };

    while (hasZero(A) || hasZero(B))
    {
      --A;
      ++B;
    }

    return vector<int>{A, B};
  }
};

File: javascript/leetcode/1317. Convert Integer to the Sum of Two No-Zero Integers.js
/**
 * @param {number} n
 * @return {number[]}
 */
var getNoZeroIntegers = function (n) {
  let A = n,
    B = 0;

  const hasZero = (num) => {
    if (!num) return true;

    while (num) {
      if (num % 10 === 0) return true;
      num = parseInt(num / 10);
    }

    return false;
  };

  while (hasZero(A) || hasZero(B)) {
    --A;
    ++B;
  }

  return [A, B];
};
